### PR TITLE
Remove premature protection for Start/Stop in jaeger receiver

### PR DESCRIPTION
**Pull request by bogdandrutu**
No other component implement this, and we should trust the service that it does the right thing.

Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

### CODE REVIEW

File: receiver/jaegerreceiver/trace_receiver.go [hunk 1]
```diff
@@ -189,51 +185,42 @@ func (jr *jReceiver) Start(_ context.Context, host component.Host) error {
...
-
-		if jr.agentServer != nil {
-			if aerr := jr.agentServer.Close(); aerr != nil {
-				errs = append(errs, aerr)
-			}
-		}
-		for _, processor := range jr.agentProcessors {
-			processor.Stop()
-		}
+	jr.mu.Lock()
```
Comment (jrcamp) [inline receiver/jaegerreceiver/trace_receiver.go:200]: Don't think this locking is needed is it? Starting and Stopping components get run in same goroutine so I assume memory ordering should be ok. I think most components depend on this memory consistency as well.

Comment (bogdandrutu) [reply receiver/jaegerreceiver/trace_receiver.go:200]: Separate issue/concern but good catch, should be removed

Comment (bogdandrutu) [reply receiver/jaegerreceiver/trace_receiver.go:200]: Do you mind doing a PR

Comment (jrcamp) [reply receiver/jaegerreceiver/trace_receiver.go:200]: #3041

**Review by jrcamp — APPROVED** [review]

**Pull request merged**